src/Polly.Shared/RetrySyntaxAsync.cs

```diff
@@ -119,9 +198,23 @@ public static Policy RetryForeverAsync(this PolicyBuilder policyBuilder, Action<
         /// <returns>The policy instance.</returns>
         public static Policy WaitAndRetryAsync(this PolicyBuilder policyBuilder, IEnumerable<TimeSpan> sleepDurations)
         {
+            return WaitAndRetryAsync(policyBuilder, false, sleepDurations);
+        }
+
+        /// <summary>
+        ///     Builds a <see cref="Policy" /> that will wait and retry as many times as there are provided
+        ///     <paramref name="sleepDurations" />
+        ///     On each retry, the duration to wait is the current <paramref name="sleepDurations" /> item.
+        /// </summary>
+        /// <param name="policyBuilder">The policy builder.</param>
+        /// <param name="continueOnCapturedContext">Whether to continue on a captured synchronization context.</param>
+        /// <param name="sleepDurations">The sleep durations to wait for on each retry.</param>
+        /// <returns>The policy instance.</returns>
+        public static Policy WaitAndRetryAsync(this PolicyBuilder policyBuilder, bool continueOnCapturedContext, IEnumerable<TimeSpan> sleepDurations)
```

Comment:
Per previous discussion, move `bool continueOnCapturedContext` to after `sleepDurations`